src/neo/SmartContract/Native/Tokens/Nep11Token.cs

```diff
@@ -0,0 +1,278 @@
+#pragma warning disable IDE0060
+
+using Neo.Cryptography;
+using Neo.IO;
+using Neo.IO.Json;
+using Neo.Ledger;
+using Neo.Persistence;
+using Neo.SmartContract.Manifest;
+using Neo.VM;
+using Neo.VM.Types;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+using Array = Neo.VM.Types.Array;
+
+namespace Neo.SmartContract.Native.Tokens
+{
+    public abstract class Nep11Token<TToken, TAccount> : NativeContract
+        where TToken : Nep11TokenState, new()
+        where TAccount : AccountState, new()
+    {
+        public override string[] SupportedStandards { get; } = { "NEP-10", "NEP-11" };
+        public abstract string Symbol { get; }
+        public abstract byte Decimals { get; }
+        public BigInteger Factor { get; }
+
+        private const byte Prefix_TotalSupply = 20;
+        private const byte Prefix_Ownership = 21;
+        private const byte Prefix_TokenId = 22;
+
+        private const int MaxTokenIdLength = 256;
```

Comment:
I think that TokenId should have a fixed size

Reply:
This is difficult because different NFT assets may have different Id systems.For example, the domain name is a string of characters, according to the "xxx.xxxx.xxxxx" format, warehouse product number SKUxxxxxxx.It is difficult to limit them to the same length, which is also the purpose of introducing innerKey in the design.

Reply:
His id could be a hash `UInt256`

Reply:
Haha, I thought about this problem.The number of bytes of UInt256 is 32 bytes, and this length is too short for the domain name, because in the domain name specification, the maximum length of a standard domain name is 67 characters.

Reply:
In addition, for example, an asset such as a domain name, the name is its Id, and then converted into a hash, which is a bit superfluous and requires additional resources to store the name

Reply:
You can have a big domain name but his id could be the hash of the domain.

Reply:
We need domain name as the `tokenId`  in nns, when we want to create a sub-domain name. And actually, we don't use `tokenId` as the storage key, we use the `innerKey`, which is `UInt160`.